core/src/ast/ast_literal.rs

```diff
@@ -26,6 +28,43 @@ impl ToSql for AstLiteral {
     }
 }
 
+impl<'a> TryFrom<Value> for AstLiteral {
+    type Error = Error;
+
+    fn try_from(value: Value) -> Result<Self> {
+        let ast_literal = match value {
+            Value::Bool(v) => AstLiteral::Boolean(v),
+            Value::I8(v) => AstLiteral::Number(BigDecimal::from_i8(v).unwrap()),
+            Value::I16(v) => AstLiteral::Number(BigDecimal::from_i16(v).unwrap()),
+            Value::I32(v) => AstLiteral::Number(BigDecimal::from_i32(v).unwrap()),
+            Value::I64(v) => AstLiteral::Number(BigDecimal::from_i64(v).unwrap()),
+            Value::I128(v) => AstLiteral::Number(BigDecimal::from_i128(v).unwrap()),
+            Value::U8(v) => AstLiteral::Number(BigDecimal::from_u8(v).unwrap()),
+            Value::F64(v) => AstLiteral::Number(BigDecimal::from_f64(v).unwrap()),
+            Value::Decimal(v) => {
+                AstLiteral::Number(BigDecimal::from_f64(v.try_into().unwrap()).unwrap())
+            }
+            Value::Str(v) => AstLiteral::QuotedString(v),
+            Value::Bytea(v) => AstLiteral::HexString(hex::encode(v)),
+            Value::Date(v) => AstLiteral::QuotedString(v.to_string()),
+            Value::Timestamp(v) => AstLiteral::QuotedString(v.to_string()),
+            // Value::Timestamp(v) => {
+            //     AstLiteral::QuotedString(DateTime::<Utc>::from_utc(v, Utc).to_string().into())
+            // }
+            Value::Time(v) => AstLiteral::QuotedString(v.to_string()),
+            Value::Interval(v) => AstLiteral::QuotedString(v.into()),
+            Value::Uuid(v) => {
+                AstLiteral::QuotedString(Uuid::from_u128(v).hyphenated().to_string().into())
+            }
+            Value::Map(v) => todo!(),
+            Value::List(v) => todo!(),
```

Comment:
We have to decide how to show Map and List with Literal

Reply:
Hmm... this really looks interesting.
What we need is generate SQL from AST so you are trying to convert `Value` to `AstLiteral`.

Reply:
Yes.. It is the first case against our parsing direction

Reply:
Currently thinking about
expanding `Expr` to contain `Value`, then we can impl `ToSql` for `Value`.